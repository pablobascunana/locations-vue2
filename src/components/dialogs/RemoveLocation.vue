<template>
  <v-dialog v-model="removeLocationDialogVisible" max-width="500px">
    <v-card>
      <v-card-title>
        {{ $t('locations.delete') }}
        <v-spacer></v-spacer>
        <v-icon aria-label="Close" @click="closeDialog()">mdi-close</v-icon>
      </v-card-title>
      <v-card-text class="pt-0 pb-0 text-center">
        <p>¿Está seguro de eliminar la localización <span class="font-weight-bold">{{ marker.title }}</span>?</p>
      </v-card-text>
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn
          color="primary"
          text
          @click="closeLocation()"
        >{{ $t('generic.buttons.exit')}}
        </v-btn>
        <v-btn
          color="primary"
          text
          @click="deleteLocation()"
        >{{ $t('generic.buttons.delete')}}
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
import Api from '@/api/services';
import { mapState } from 'vuex'
export default {
  name: 'removeLocation',
  props: {
    marker: { type: Object, default: () => {}},
    userUuid: { type: String, required: true }
  },
  computed: {
    ...mapState('application', ['removeLocationDialogVisible'])
  },
  methods: {
    async deleteLocation() {
      try {
        let { data } = await Api.deleteLocation(this.userUuid, this.marker);
        this.$emit('updateMarkers', data.markers);
        this.closeDialog();
      } catch (error) {
        this.EventBus.$emit('showSnackBar', { 
          snackBarMessage: this.$t('locations.error.deleteGeneric'),
          snackBarColor: 'error'
        });
        this.closeDialog();
      }
    },
    closeDialog() {
      this.$store.commit('application/isRemoveLocationDialogVisible', false);
    }
  }
}
</script>

<style>

</style>