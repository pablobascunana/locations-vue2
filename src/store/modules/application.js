function initialApplicationState() {
  return {
    isLogin: true,
    accessToken: undefined,
    refreshToken: undefined,
    addLocationDialogVisible: false,
    removeLocationDialogVisible: false
  }
}

const application = {
  namespaced: true,
  state: initialApplicationState,
  mutations: {
    resetApplicationStore(state) {
      const initState = initialApplicationState();
      Object.keys(initState).forEach(key => {
        state[key] = initState[key];
      })
    },
    isLogin(state, visible) {
      state.isLogin = visible;
    },
    setAccessToken(state, token) {
      state.accessToken = `Bearer ${token}`;
    },
    setRefreshToken(state, token) {
      state.refreshToken = token;
    },
    isAddLocationDialogVisible(state, visible) {
      state.addLocationDialogVisible = visible;
    },
    isRemoveLocationDialogVisible(state, visible) {
      state.removeLocationDialogVisible = visible
    }
  }
};

export default application;
  