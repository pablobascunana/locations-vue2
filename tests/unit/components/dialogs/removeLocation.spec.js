import { createLocalVue, shallowMount } from '@vue/test-utils'

import Vuetify from 'vuetify'
import Vuex from 'vuex'

import application from '@/store/modules/application.js'

import RemoveLocation from '@/components/dialogs/RemoveLocation.vue'

import * as historicalLocationsMock from '../../mocks/historicalLocations.mock.js'

const Plugins = {
    install(Vue) {
      Vue.prototype.EventBus = new Vue();
    }
};

const localVue = createLocalVue();
localVue.use(Plugins);
localVue.use(Vuex);
let vuetify = new Vuetify();

describe('RemoveLocation', () => {
    let wrapper;
    let store;
    let state = {};
    let actions = {};
    let modules = {
        application
    };
    const mocks = {
        EventBus: {
            $emit: jest.fn(),
        },
        $t: () => {'some specific text in any language'}
    };
    beforeEach(() => {
        store = new Vuex.Store({ state, actions, modules });

        wrapper = shallowMount(RemoveLocation, {
            localVue,
            vuetify,
            store,
            propsData: {
                marker: { type: Boolean, default: historicalLocationsMock.MARKER }
            },
            mocks
        });
    });

    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()    
    });

    it('Renders a vue instance', () => {
        expect(wrapper.isVueInstance()).toBe(true);
    });


    it('Call to delete location function', () => {
        const deleteLocationFn = jest.spyOn(wrapper.vm, 'deleteLocation');
        wrapper.vm.deleteLocation();
        expect(deleteLocationFn).toBeCalled();
    });

    it('Call to update markers and close dialog function', () => {
        const updateMarkersAndCloseDialogFn = jest.spyOn(wrapper.vm, 'updateMarkersAndCloseDialog');
        wrapper.vm.updateMarkersAndCloseDialog();
        expect(updateMarkersAndCloseDialogFn).toBeCalled();
    });

    it('Call to show error snackBar and close dialog function', () => {
        const showErrorSnackBarAndCloseDialogFn = jest.spyOn(wrapper.vm, 'showErrorSnackBarAndCloseDialog');
        wrapper.vm.showErrorSnackBarAndCloseDialog();
        expect(showErrorSnackBarAndCloseDialogFn).toBeCalled();
    });
    
    it('Call to close dialog function', () => {
        const closeDialogFn = jest.spyOn(wrapper.vm, 'closeDialog');
        wrapper.vm.closeDialog();
        expect(closeDialogFn).toBeCalled();
        expect(wrapper.vm.$store.state.application.addLocationDialogVisible).toBe(false);
    });
});