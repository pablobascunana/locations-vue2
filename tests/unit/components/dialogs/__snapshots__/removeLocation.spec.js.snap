// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveLocation Should match snapshot 1`] = `
<v-dialog max-width="500px">
  <v-card>
    <v-card-title>

      <v-spacer></v-spacer>
      <v-icon aria-label="Close">mdi-close</v-icon>
    </v-card-title>
    <v-card-text class="pt-0 pb-0 text-center">
      <p><span class="font-weight-bold"> </span>?</p>
    </v-card-text>
    <v-card-actions>
      <v-spacer></v-spacer>
      <v-btn color="primary" text="">
      </v-btn>
      <v-btn color="red" text="">
      </v-btn>
    </v-card-actions>
  </v-card>
</v-dialog>
`;

exports[`RemoveLocation should match snapshot 1`] = `
<v-dialog max-width="500px">
  <v-card>
    <v-card-title>

      <v-spacer></v-spacer>
      <v-icon aria-label="Close">mdi-close</v-icon>
    </v-card-title>
    <v-card-text class="pt-0 pb-0 text-center">
      <p><span class="font-weight-bold"> </span>?</p>
    </v-card-text>
    <v-card-actions>
      <v-spacer></v-spacer>
      <v-btn color="primary" text="">
      </v-btn>
      <v-btn color="red" text="">
      </v-btn>
    </v-card-actions>
  </v-card>
</v-dialog>
`;
